File: Sources/FocusCore/DynamicCard.swift
import Foundation

public struct DynamicCard: Identifiable, Equatable, Sendable {
    public enum Kind: String, Equatable, Sendable {
        case text
        case image
        case video
        case articleLike
    }

    public struct Author: Equatable, Sendable {
        public let name: String
        public let avatarURL: URL?

        public init(name: String, avatarURL: URL?) {
            self.name = name
            self.avatarURL = avatarURL
        }
    }

    public let id: String
    public let kind: Kind
    public let author: Author
    public let publishTime: String
    public let text: String
    public let coverURLs: [URL]
    public let targetURL: URL
    public let videoURL: URL?

    public init(
        id: String,
        kind: Kind,
        author: Author,
        publishTime: String,
        text: String,
        coverURLs: [URL],
        targetURL: URL,
        videoURL: URL?
    ) {
        self.id = id
        self.kind = kind
        self.author = author
        self.publishTime = publishTime
        self.text = text
        self.coverURLs = coverURLs
        self.targetURL = targetURL
        self.videoURL = videoURL
    }
}

File: focus-core/src/commonMain/kotlin/org/bilibilifocus/core/model/DynamicCard.kt
package org.bilibilifocus.core.model

import kotlinx.serialization.Serializable

@Serializable
data class DynamicCard(
    val id: String,
    val kind: Kind,
    val author: Author,
    val publishTime: String,
    val text: String,
    val coverURLs: List<String>,
    val targetURL: String,
    val videoURL: String? = null,
) {
    @Serializable
    enum class Kind {
        TEXT,
        IMAGE,
        VIDEO,
        ARTICLE_LIKE,
    }

    @Serializable
    data class Author(
        val name: String,
        val mid: Long = 0,
        val avatarURL: String? = null,
    )
}
